### ISSUE TITLE
[Repository] RQL/XML auto complete

### ISSUE THREAD

**Issue by troussej** [jc7447/BetterDynAdmin#104]
When working on autocomplete for Dash, I started playing with [jquery-textcomplete](https://github.com/yuku-t/jquery-textcomplete)

It's quite easy to use, so maybe we can add it to the xml editor on repositories.

**Comment by troussej** [comment]
Textcomplete does not work at all with CM, however CM has the show-hint module + xml that works quite well -> pushed to dev

The only issue it that some events inside CM are not sent/catched correctly with FF + GreaseMonkey (it works on FF alone - I made a [jsfiddle](http://jsfiddle.net/troussej/yebj2zo2/) )  so I added some hacks when the browser is FF. It solves the issue but the navigation is a bit clunky.

The other point is that for it to work I added the xml mode on CodeMirror, which makes the xml textbox behave a bit differently (auto-indent + other stuff) - I'm not sure we want to keep that.

All of this is isolated in the code so we can switch it off if necessary.

**Comment by jc7447** [comment]
It would very cool to have the autocomplete feature for the RQL syntax like (ALL, ORDER BY, RANGE, ...)

**Comment by troussej** [comment]
Yeah I tried at the time but it's only using the built in xml complete which only completes nodes and attributes. 

Btw i'm not working on ATG right now so I'm not gonna work much on BDA

**Comment by troussej** [comment]
should we close it for now?
